### PR TITLE
Add API management for issue/pull and comment attachments

### LINKED ISSUE #3690: Issue API and Attachments
- Gitea version (or commit ref): 1.3.3
- Git version: n/a
- Operating system:n/a
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [x ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [] Yes (provide example URL)
  - [ ] No
  - [ a] Not relevant
- Log gist:

## Description
The Issues API doesn't expose attachments, is this by design?
...

### PR DESCRIPTION
**Pull request by KN4CK3R**
Close #14601
Fix #3690

Revive of #14601.
Updated to current code, cleanup and added more read/write checks.

### CODE REVIEW

**Comment by mattrpav** [comment]
@KN4CK3R thanks for jumping onto this feature!

File: routers/api/v1/repo/issue_attachment.go [hunk 1]
```diff
@@ -0,0 +1,373 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
```
Comment (lunny) [inline routers/api/v1/repo/issue_attachment.go]: Please update the copyright header

**Review by zeripath — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
make L-G-T-M work

File: modules/structs/issue.go [hunk 2]
```diff
@@ -41,18 +41,19 @@ type RepositoryMeta struct {
...
+	URL              string        `json:"url"`
+	HTMLURL          string        `json:"html_url"`
+	Index            int64         `json:"number"`
+	Poster           *User         `json:"user"`
+	OriginalAuthor   string        `json:"original_author"`
+	OriginalAuthorID int64         `json:"original_author_id"`
+	Title            string        `json:"title"`
+	Body             string        `json:"body"`
+	Ref              string        `json:"ref"`
+	Attachments      []*Attachment `json:"assets"`
```
Comment (wxiaoguang) [inline modules/structs/issue.go:54]: A little curious, why it's called `assets` in JSON (and route path), it introduces unnecessary inconsistency IMO.

**Pull request merged**